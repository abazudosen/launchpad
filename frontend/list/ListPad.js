/* @flow */

import React, { Component } from 'react';
import type { Pad } from './types';
import { getPrismHighlighter } from '../services/CodeService';
import './ListPad.less';

type ListPadProps = {|
  pad: Pad,
|};

export default class ListPad extends Component {
  props: ListPadProps;

  state = ({
    code: null,
  }: {
    code: ?string,
  });

  componentDidMount() {
    this.hilightCode();
  }

  componentDidUpdate(nextProps: ListPadProps) {
    if (nextProps.pad.code !== this.props.pad.code) {
      this.setState({
        code: null,
      });
      this.hilightCode();
    }
  }

  async hilightCode() {
    const code = (await getPrismHighlighter())(
      this.props.pad.code,
      'javascript',
    );
    this.setState({
      code,
    });
  }

  renderCode() {
    if (this.state.code) {
      return (
        <div
          className="ListPad-Code"
          dangerouslySetInnerHTML={{
            __html: this.state.code,
          }}
        />
      );
    } else {
      return null;
    }
  }

  render() {
    const pad = this.props.pad;
    const padUser = pad.user || {};
    return (
      <div className="ListPad">
        <a className="ListPad-Link" href={`/${this.props.pad.id}`}>
          <div className="ListPad-Header">
            <div className="ListPad-HeaderUsername">
              {padUser.githubUsername || 'Unknown'}
            </div>
            <div className="ListPad-HeaderTitle">
              {this.props.pad.title || '(Untitled)'}
            </div>
          </div>
          {this.renderCode()}
        </a>
      </div>
    );
  }
}
