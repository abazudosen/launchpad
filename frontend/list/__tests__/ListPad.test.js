/* @flow */

import React from 'react';
import ListPad from '../ListPad';
import renderer from 'react-test-renderer';

describe('Pad List Component', () => {
  test('renders normal', () => {
    const component = renderer.create(
      <ListPad
        pad={{
          id: 'fakeid1',
          title: 'Fake 1',
          description: 'Description 1',
          code: 'code',
          deployedCode: 'deployed code',
          url: 'http://example.com',
          user: {
            id: 'test-ids',
            githubUsername: 'example',
            pads: [],
          },
          context: [
            {
              key: 'secret',
              value: 'secret',
            },
          ],
          dependencies: [
            {
              name: 'graphql-js',
              version: '0.8.9',
            },
          ],
          defaultQuery: '{ hello }',
          defaultVariables: null,
          token: 'test-token-2',
          draft: null,
        }}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
