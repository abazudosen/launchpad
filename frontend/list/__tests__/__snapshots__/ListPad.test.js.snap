// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pad List Component renders normal 1`] = `
<div
  className="ListPad"
>
  <a
    className="ListPad-Link"
    href="/fakeid1"
  >
    <div
      className="ListPad-Header"
    >
      <div
        className="ListPad-HeaderUsername"
      >
        example
      </div>
      <div
        className="ListPad-HeaderTitle"
      >
        Fake 1
      </div>
    </div>
  </a>
</div>
`;
